source/analysis/AM1DDerivativeAB.cpp

```diff
@@ -301,15 +296,15 @@ bool AM1DDerivativeAB::values(const AMnDIndex &indexStart, const AMnDIndex &inde
 				badIndices.append(totalSize-1);
 
 			// Compute all the values
-			outputValues[0] = (data.at(1)-data.at(0))/(axis.at(1)-axis.at(0));
-			outputValues[totalSize-1] = (data.at(totalSize-1)-data.at(totalSize-2))/(axis.at(totalSize-1)-axis.at(totalSize-2));
+			cachedOutput_[0] = (data.at(1)-data.at(0))/(axis.at(1)-axis.at(0));
```

Comment:
This part seems to be confusing me a little bit (not specifically what you've done, but the already present checks for divide by zero).
If `axis[0] == axis[1]` (for example) the check finds this and adds 0 to the badIndices list. However we still then proceed to 
```C++
cachedOutput_[0] = (data.at(1)-data.at(0))/(axis.at(1)-axis.at(0));
```
which would surely still cause a divide by zero error, no? The inherent imprecision in doubles is part of what's confusing me here.